### DISCUSSION TITLE
Setting up relationships

### DISCUSSION THREAD

**Discussion by liorbalmas** [abpframework/abp#7284] [Q&A] [answered]
Hi, 
DDD forbids navigation properties and that's the recommended practice in docs.
On the other hand, EF uses navigation properties to set up relationships by convention.
Does that mean relationships should be explicitly coded in *ModelCreatingExtentions class?

Thank you

**Comment by realLiangshiwei** [comment, marked as answer]
See https://docs.abp.io/en/abp/latest/Domain-Driven-Design-Implementation-Guide#discussion-about-the-database-independence-principle

**Reply by liorbalmas** [reply]
According to that, the answer is yes.
Since the framework is based on DDD, I think it's a good idea to have in docs the Fluent API code demonstrating relationships, since that's not so straightforward in EF. (Compared to conventions)
Example:

```
builder.Entity<Child>(b =>
            {
                b.ToTable(KidsConsts.DbTablePrefix + "Childs",
                          KidsConsts.DbSchema);
                b.ConfigureByConvention(); 
                b.HasOne<Parent>()
                .WithMany()
                .HasForeignKey(c => c.Parent1Id);
                b.HasOne<Parent>()
               .WithMany()
               .HasForeignKey(c => c.Parent2Id);

            });
```
